sql/schemachanger: Remove feature gate for ALTER COLUMN TYPE

spilchen opened the pull request:
Previously, altering a column’s type requiring a backfill required the enable_experimental_alter_column_type_general setting to be enabled. This restriction has been removed, as full support for ALTER COLUMN TYPE has been implemented in the Declarative Schema Changer (DSC).

The setting is still checked in mixed-version clusters, as the operation relies on 25.1 dependency rules, and when using the legacy schema changer.

Epic: CRDB-25314
Informs #49329
Release note (sql change): Altering a column’s type no longer requires enabling the enable_experimental_alter_column_type_general setting. This change makes the feature generally available.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135936)
<!-- Reviewable:end -->

rafiss reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135936)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @herkolategan, @nameisbhaskar, and @spilchen)

___
*[`-- commits` line 15 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135936#-OCHC8td4oFFMiLhvslN:-OCHC8td4oFFMiLhvslO:bn2g40z):*
super nit: let's use `sql change` for the release note category, as schema changes and session settings both fall under that.


<!-- Sent from Reviewable.io -->

spilchen committed:
sql/schemachanger: Remove feature gate for ALTER COLUMN TYPE

Previously, altering a column’s type requiring a backfill required the
enable_experimental_alter_column_type_general setting to be enabled.
This restriction has been removed, as full support for ALTER COLUMN TYPE
has been implemented in the Declarative Schema Changer (DSC).

The setting is still checked in mixed-version clusters, as the operation
relies on 25.1 dependency rules, and when using the legacy schema
changer.

Epic: CRDB-25314
Informs #49329
Release note (sql change): Altering a column’s type no longer requires
enabling the enable_experimental_alter_column_type_general setting. This
change makes the feature generally available.

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135936)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @herkolategan, @nameisbhaskar, and @rafiss)

___
*[`-- commits` line 15 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135936#-OCHC8td4oFFMiLhvslN:-OCImwEhBnAPkULgZJQN:b-896fix):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

super nit: let's use `sql change` for the release note category, as schema changes and session settings both fall under that.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+
